### ISSUE TITLE
It is impossible to clear the relations by sending PATCH request to relationship link.

### ISSUE THREAD

**Issue by gast** [tuyakhov/yii2-json-api#41]
Hi!

If I send a null or an empty array as a data member like:

```
PATCH /articles/1/relationships/author HTTP/1.1
Content-Type: application/vnd.api+json
Accept: application/vnd.api+json

{
	"data": null // or []
}
```

1. JsonParser::[parseRelationships](https://github.com/tuyakhov/yii2-json-api/blob/a42e222fb1c263f494ceee81b00fc85db98b8fc2/src/JsonApiParser.php#L112)() will not process an empty relationship data.
2. Action::[linkRelations](https://github.com/tuyakhov/yii2-json-api/blob/a42e222fb1c263f494ceee81b00fc85db98b8fc2/src/JsonApiParser.php#L112)(), for obvious reasons, will not clear the relationship.

Is this normal behavior? Because in the JSON API specification I read that:

> 	The PATCH request MUST include a top-level member named data containing one of:
> 
> 	- a resource identifier object corresponding to the new related resource.
> 	- **null**, to remove the relationship.
> 	
> 	....
> 	
> 	For all request types, the body MUST contain a data member whose value is an **empty array** or an array of resource identifier objects.
> 	
> 	....

	
[jsonapi.org](http://jsonapi.org/format/#crud-updating-relationships)

	
Thanks for this extension!

**Comment by tuyakhov** [comment]
Hi @gast!
Thank you for reporting this. It seems like a bug to me. I will investigate further and provide a solution soon.

**Comment by tuyakhov** [comment]
Resolved in #43
